app/lib/api/file_system/file_system_remote.dart

```diff
@@ -143,10 +143,30 @@ mixin RemoteSystem {
   }
 
   Future<void> clearCachedContent() async {
-    var cacheDir = await getRemoteCacheDirectory();
-    var directory = Directory(cacheDir);
-    if (await directory.exists()) {
-      await directory.delete(recursive: true);
+    try {
+      var cacheDir = await getRemoteCacheDirectory();
+      var directory = Directory(cacheDir);
+      final exists = await directory.exists();
+      if (exists) {
+        try {
+          await directory.delete(recursive: true);
+          // Directory deleted successfully, perform any needed actions
+        } on FileSystemException catch (e) {
+          if (e.osError?.errorCode == 32) {
+            // Directory in use, retry after a short delay
+            await Future.delayed(const Duration(seconds: 1));
+            await clearCachedContent();
```

Comment:
Here I wouldn't add recursive since it can result in a stack overflow. Maybe better a for/while loop with a limit of 5?

Reply:
I didn't understand before, you're right. I'll edit it